### DISCUSSION TITLE
How much of my code does CodeCarbon get access to?

### DISCUSSION THREAD

**Discussion by hayley-wragg** [mlco2/codecarbon#501] [Q&A] [answered]
We want to set up a carbon tracking script for lot's of staff in our department. But we need to make sure that the only information being shared is the energy/ carbon usage.
We need to ensure the code and any data called within the code is not shared with CodeCarbon, as well as any identifiable information on the machine.

**Comment by LuisBlanche** [comment, marked as answer]
Hello, 
First of all, it depends on how you use code carbon. You can for instance use offline mode, which would be 100% private, and would store the results of your C02 monitoring in a CSV on the machine that is running the code. 
If you choose however to send data to the publicly hosted API, and to the [dashboard](https://dashboard.codecarbon.io/), then this data is public. 
You can also choose to deploy the API, and Database  on your onw infastructur (on prem or cloud) which you can do using the [docker-compose ](https://github.com/mlco2/codecarbon/blob/master/docker-compose.yml)

You can report to the [example](https://github.com/mlco2/codecarbon/tree/master/examples) folder to see how the different configuration log data. In this very folder you have an [emissions.csv](https://github.com/mlco2/codecarbon/blob/master/examples/emissions.csv) file that shows what type of data is logged.

It is on our 2024 roadmap to improve the API and Dashboard to include users and organisations management to allow secure control over what is shared or not.
